imeplusplus/dont-panic-api
# Add CRUD operations for Problems

## Changes

This PR adds CRUD (CREATE, READ - or 'GET', UPDATE, and DELETE) basic functionality for 'Problem' vertices. Its implementation is VERY similar to other vertices that already exist so we can stay consistent.
- Add `model/problem.go`, `dbOperations/problems.go` and `handler/problems.go`;
- Add routing for the operations

## How to test it

1. `go build .` and `./dont-panic-api` with a valid Gremlin database configuration.
2. Open postman, and try the following routes to test each feature:

 
- `CREATE`
 It's a `POST` request with URL `localhost:8080/api/problems`. Remember to put a valid body. Example:
 ```json
 {
   "name": "xavi-and-the-cookies",
   "subjects": [
     "dynamic-programming",
     "math"
   ],
   "difficulty": 2
 }
 ```
  The expected result of this request is that the Problem will be created if there isn't already any other Problem with the name you specified.
 
- `READ`
 To fetch all problems, make a `GET` request with URL `localhost:8080/api/problems`. To get a specific Problem, make a GET request with URL `localhost:8080/api/problems/{name}`, being `{name}` the name of the Problem.
 The expected result is a `json` with the information from the problems you requested.
 
- `UPDATE`
 It's a `PUT` request with URL `localhost:8080/api/problems/{name}`, being `{name}` the name of the Problem. Put a valid body that represents the updated Problem.
 The expected result is that the Problem information will be updated if there exists a Problem in the database with the name you specified.
 
- `DELETE`
 It's a `DELETE` request with URL `localhost:8080/api/problems/{name}`, being `{name}` the name of the Problem.
 The expected result is that the Problem will be delete if there is a Problem in the database with the name you specified, or an error if there isn't.
